src/containers/PageBuilder/Primitives/Heading/Heading.module.css

```diff
@@ -6,24 +6,93 @@
 .h5,
 .h6 {
   margin-top: 0;
+  margin-bottom: 0;
   line-height: 1.33;
   font-weight: bold;
+  color: var(--matterColorAlmostDark);
 }
 
 /* Specific styles */
 .h1 {
   font-size: 40px;
+
+  /* If ´& + *´ if used, margin-top needs !important to overwrite */
+  & + p,
+  & + a,
+  & + ul,
+  & + ol,
+  & + code,
+  & + div,
+  & + h1,
+  & + h2,
+  & + h3,
+  & + h4,
+  & + h5,
+  & + h6 {
```

Comment:
In what situation, the margin-top is not used?

Could we turn this around? `margin-top: 24px;` is the default and it's not there in certain situations.

Reply:
I found this:
https://developer.mozilla.org/en-US/docs/Web/CSS/:only-child

So, maybe this could be written as:

```
.h1 {
  font-size: 40px;
  margin: 24px;

  &:only-child {
    margin: 0px;
  }
}
```

And the same approach to other situations too.

Reply:
A primitive shouldn't have any whitespace around it on it's own, because that messes up the vertical spacing. Few examples,

- Sections always have margin-y of 64px.
- H1 has a margin-bottom of 24px. Section only has a H1, but the margin-bottom of the H1 gets added to the section's margin-bottom, which causes too much whitespace below the H1, 88px which doesn't match with the margin-top. Same thing with primitive default margin-tops, too much whitespace.
- Headings followed by another headings might require more margin-bottom than for example paragraphs or buttons
- So the common solution is to use sibling selectors and instead of building the margins down for each primitive, we build margins up after the first primitive per the whitespace relationship between primitives.

The infamous :has selector which I've been trying to get your opinion on solves all the pain for "context-aware" margins between primitives. Not sure if :only-child will do the trick, but I'll try!